backend/inspirehep/records/api/base.py

```diff
@@ -176,6 +176,17 @@ def get_record(cls, record_uuid, with_deleted=False, record_version=None):
             raise NoResultFound
         return record
 
+    @classmethod
+    def get_records_batched(cls, ids, with_deleted=False, max_batch=100):
+        for batch in chunker(ids, max_chunk_size=max_batch):
+            query = cls.model_cls.query.filter(cls.model_cls.id.in_(batch))
+            if not with_deleted:
+                query = query.filter(
+                    cls.model_cls.json.op("->")("deleted") == None  # noqa
+                )
```

Comment:
this is incorrect, `deleted = False` is allowed in a record. You need to check that `delete: True` doesn't occur.

Reply:
ok, thank you